addons/sale/models/product_template.py

```diff
@@ -294,10 +307,9 @@ def _is_add_to_cart_possible(self, parent_combination=None):
         :rtype: bool
         """
         self.ensure_one()
-        if not self.active or not self.sale_ok:
-            # for performance: avoid calling `_get_possible_combinations`
-            return False
```

Comment:
Keep the diff minimal to avoid conflicts and avoid losing the blame of the code ;)
```python
if not self._can_be_added_to_cart():
    # for performance: avoid calling `_get_possible_combinations`
    return False
return next(self._get_possible_combinations(parent_combination), False) is not False

Reply:
My bad, thanks :+1: 
And what about that ? (just for my knowledge :wink:) :
```python
if self.active and not self._can_be_added_to_cart(): # in _is_add_to_cart_possible
return self.sale_ok # in main _can_be_added_to_cart
return super()._can_be_added_to_cart() or self.rent_ok  # in override _can_be_added_to_cart
```

Reply:
It's even better :+1:

Reply:
Perfect, thanks :+1: